File: ios/PouleParty/Models/PricingConfig.swift
//
//  PricingConfig.swift
//  PouleParty
//

import Foundation

struct PartyPlansConfig: Codable, Equatable {
    var free: FreePlan = FreePlan()
    var flat: FlatPlan = FlatPlan()
    var deposit: DepositPlan = DepositPlan()

    struct FreePlan: Codable, Equatable {
        var maxPlayers: Int = 5
        var enabled: Bool = true
    }

    struct FlatPlan: Codable, Equatable {
        var pricePerPlayerCents: Int = 300
        var minPlayers: Int = 6
        var maxPlayers: Int = 50
        var enabled: Bool = true
    }

    struct DepositPlan: Codable, Equatable {
        var depositAmountCents: Int = 1000
        var commissionPercent: Double = 15.0
        var enabled: Bool = true
    }
}

File: android/app/src/main/java/dev/rahier/pouleparty/model/PricingConfig.kt
package dev.rahier.pouleparty.model

data class PartyPlansConfig(
    val free: FreePlan = FreePlan(),
    val flat: FlatPlan = FlatPlan(),
    val deposit: DepositPlan = DepositPlan()
) {
    data class FreePlan(
        val maxPlayers: Int = 5,
        val enabled: Boolean = true
    )

    data class FlatPlan(
        val pricePerPlayerCents: Int = 300,
        val minPlayers: Int = 6,
        val maxPlayers: Int = 50,
        val enabled: Boolean = true
    )

    data class DepositPlan(
        val depositAmountCents: Int = 1000,
        val commissionPercent: Double = 15.0,
        val enabled: Boolean = true
    )
}
